code/game/objects/items/blueprints.dm

```diff
@@ -193,13 +193,15 @@
 			else
 				usr << "<span class='warning'>Error! Please notify administration.</span>"
 				return
+
 	var/list/turfs = res
 	var/str = trim(stripped_input(usr,"New area name:", "Blueprint Editing", "", MAX_NAME_LEN))
 	if(!str || !length(str)) //cancel
 		return
 	if(length(str) > 50)
 		usr << "<span class='warning'>The given name is too long.  The area remains undefined.</span>"
 		return
+	var/area/old = get_area(get_turf(src))
```

Comment:
var/area/old = get_area(get_turf(src))
var/oldgravity = old.has_gravity

and then later....
A.has_gravity = oldgravity